### DISCUSSION TITLE
About Song IDs

### DISCUSSION THREAD

**Discussion by agonzalo36r** [AzuraCast/AzuraCast#5545] [Q&A]
Hi!

I was testing the new export feature, and on the csv there's an ID column, whose values don't match with the song id on the Now Playing API. However, these ID column values match the first part of the cover art filenames of the songs on the API. 
I don't know if there's something wrong with my installation or if these IDs are not the same. In that case, is there a way to obtain the same ID listed in the csv from the API?

Thanks a lot for taking the time to read this!

**Comment by BusterNeece** [comment]
@agonzalo36r Moved this over to a Discussion since it's not really reporting a bug.

There are two types of IDs that you'll often see associated with media:
 - The `media_id`, which is a unique identifier we use to correlate to a specific row in the `StationMedia` database, referring to a specific track in a specific storage location, and
 - The `song_id`, which is a hash based on the song's artist and title combined with special characters stripped out, and is used to match songs currently being played by Shoutcast/Icecast with records in our database.

The bulk CSV media importer/exporter uses the unique Media ID, so that we can match with an exact record in an exact location. The song ID is often seen in API reports as we used to use it quite a bit more frequently to identify songs (and to join them against a single monolithic `Songs` table, which we've since removed on account of how huge it would get for large stations and how little purpose it served).

If you're interested in determining how to calculate the song ID hash yourself, or need to do so for internal purposes, the code used to generate the hash is here, and is rather simple to implement in most languages: https://github.com/AzuraCast/AzuraCast/blob/d0da31846461e209aa7a0d71195243e08ebf7a43/src/Entity/Song.php#L27-L67
